src/redux/actions/samples/createSample.js

```diff
@@ -66,17 +62,17 @@ const createSample = (
 
     throwIfRequestFailed(response, json, endUserMessages.ERROR_SAVING);
 
-    dispatch({
+    await dispatch({
       type: SAMPLES_CREATE,
       payload: { sample: newSample, experimentId },
     });
 
-    dispatch(saveExperiment(experimentId));
+    return newSampleUuid;
   } catch (e) {
     let { message } = e;
     if (!isServerError(e)) {
       console.error(`fetch ${url} error ${message}`);
-      message = endUserMessages.ERROR_SAVING;
+      message = `${endUserMessages.ERROR_SAVING}. Please reupload sample '${name}'`;
```

Comment:
Does it make sense to perhaps move this whole message to a new `endUserMessages` element called `ERROR_SAVING_SAMPLE_CREATION` or sometihng liek that?

Reply:
Hmmm... sure